tools/check_format.py

```diff
@@ -146,6 +147,29 @@
 # yapf: enable
 
 
+# Map a line transformation function across each line of a file.
+# .bak temporaries.
+def replaceLines(path, line_xform):
+  # We used to use fileinput in the older Python 2.7 script, but this doesn't do
+  # inplace mode and UTF-8 in Python 3, so doing it the manual way.
+  with open(path, encoding='utf-8', mode='r') as f:
+    output_lines = [line_xform(line) for line in f]
+  with open(path, encoding='utf-8', mode='w') as f:
+    f.write(''.join(output_lines))
+
+
+# Obtain all the lines in a given file.
+def readLines(path):
+  with open(path, encoding='utf-8', mode='r') as f:
+    return f.readlines()
+
+
+# Read a UTF-8 encoded file as a str.
+def readFile(path):
```

Comment:
consider use pathlib (which is standard library in python3) and https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text?

Reply:
Good idea, done.